internal/db/db.go

```diff
@@ -0,0 +1,84 @@
+package db
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+)
+
+type ChatInfo struct {
+	ChatID int64
+}
+
+type DB struct {
+	path     string
+	chatsIDs map[int64]ChatInfo
+}
+
+// NewDB open and load db if exist, or create new one if not exist
+func NewDB(path string) (DB, error) {
+	if err := os.MkdirAll(path, 0777); err != nil {
+		return DB{}, err
+	}
+
+	chatsIDs := make(map[int64]ChatInfo)
+	path = filepath.Join(path, "db.json")
+
+	data, err := os.ReadFile(path)
+	if os.IsNotExist(err) || len(data) == 0 {
+		return DB{
+			path:     path,
+			chatsIDs: chatsIDs,
+		}, nil
+	}
+	if err != nil {
+		return DB{}, err
+	}
+
+	if err := json.Unmarshal(data, &chatsIDs); err != nil {
+		return DB{}, err
+	}
+
+	return DB{
+		path:     path,
+		chatsIDs: chatsIDs,
+	}, nil
+}
+
+func (db *DB) Save() error {
+	file, err := json.MarshalIndent(db.chatsIDs, "", "    ")
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(db.path, file, 0777)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (db *DB) Get(key int64) ChatInfo {
+	val, ok := db.chatsIDs[key]
+	if !ok {
+		return ChatInfo{}
+	}
```

Comment:
no need for this check because if it doesn't exist `val` would be the zero value of `ChatInfo{}`